snapshot_dbg_cli_tests/test_cli_services.py

```diff
@@ -0,0 +1,540 @@
+    self.http_service_class_mock = self.http_service_mock_patcher.start()
+    self.http_service_mock = self.http_service_class_mock.return_value
+    self.addCleanup(self.http_service_mock_patcher.stop)
+
+    self.permissions_service_patcher = patch(
+        'snapshot_dbg_cli.cli_services.PermissionsRestService', autospec=True)
+    self.permissions_service_class_mock = \
+        self.permissions_service_patcher.start()
+    self.permissions_service_mock = \
+        self.permissions_service_class_mock.return_value
+    self.addCleanup(self.permissions_service_patcher.stop)
+
+    self.firebase_management_service_patcher = patch(
+        'snapshot_dbg_cli.cli_services.FirebaseManagementRestService',
+        autospec=True)
+    self.firebase_management_service_class_mock = \
+        self.firebase_management_service_patcher.start()
+    self.firebase_management_service_mock = \
+        self.firebase_management_service_class_mock.return_value
+    self.addCleanup(self.firebase_management_service_patcher.stop)
+
+    self.firebase_rtdb_service_patcher = patch(
+        'snapshot_dbg_cli.cli_services.FirebaseRtdbRestService', autospec=True)
+    self.firebase_rtdb_service_class_mock = \
+        self.firebase_rtdb_service_patcher.start()
+    self.firebase_rtdb_service_mock = \
+        self.firebase_rtdb_service_class_mock.return_value
+    self.addCleanup(self.firebase_rtdb_service_patcher.stop)
+
+    self.snapshot_debugger_rtdb_service_patcher = patch(
+        'snapshot_dbg_cli.cli_services.SnapshotDebuggerRtdbService',
+        autospec=True)
+    self.snapshot_debugger_rtdb_service_class_mock = \
+        self.snapshot_debugger_rtdb_service_patcher.start()
+    self.snapshot_debugger_rtdb_service_mock = \
+        self.snapshot_debugger_rtdb_service_class_mock.return_value
+    self.addCleanup(self.snapshot_debugger_rtdb_service_patcher.stop)
+
+  def test_constructor_works_as_expected(self):
+    args = argparse.Namespace()
+
+    self.gcloud_service_mock.config_get_account = MagicMock(
+        return_value='foo@bar.com')
+    self.gcloud_service_mock.config_get_project = MagicMock(
+        return_value='cli-test-project')
+    self.gcloud_service_mock.get_access_token = MagicMock(
+        return_value='cli-test-access-token')
+
+    expected_is_debug_enabled = False
+
+    cli_services = CliServices(args)
+
+    self.assertEqual(args, cli_services.args)
+    self.assertEqual('foo@bar.com', cli_services.account)
+    self.assertEqual('cli-test-project', cli_services.project_id)
+    self.assertEqual('cli-test-access-token', cli_services.access_token)
+    self.assertEqual(self.http_service_mock, cli_services.http_service)
+    self.assertEqual(self.permissions_service_mock,
+                     cli_services.permissions_service)
+    self.assertEqual(self.firebase_management_service_mock,
+                     cli_services.firebase_management_service)
+
+    self.data_formatter_class_mock.assert_called_once()
```

Comment:
I'm not sure how I feel about this one.  Is it just making sure that a constructor was called?

Reply:
Comments make it clear now.